src/app/components/header/login/login.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  Validators,
+} from '@angular/forms';
+import { Router } from '@angular/router';
+import { AuthService } from 'src/app/test/auth.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css'],
+})
+export class LoginComponent implements OnInit {
+  loginForm!: FormGroup;
+
+  constructor(
+    private fb: FormBuilder,
+    private auth: AuthService,
+    private router: Router
+  ) {}
+
+  ngOnInit(): void {
+    this.loginForm = this.fb.group({
+      username: ['', Validators.required],
+      password: ['', Validators.required],
+    });
+  }
+
+  onSubmit() {
+    console.log(this.loginForm.value);
+    this.auth.login(this.loginForm.value).subscribe({
+      next: (res) => {
+        alert(res.message);
```

Comment:
here you should check if you have a valid response - then proceed to the list page - otherwise throw a login error

Reply:
But I have disabled login button if username and password entered is not valid on FE side only. So the responses will be in valid format.